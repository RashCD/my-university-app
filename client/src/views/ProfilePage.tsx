import { RouteComponentProps } from '@reach/router';
import AppHeader from '../components/AppHeader';
import CTAButton from '../components/CTAButton';
import Cookie from '../util/cookie';
import Styles from '../assets/styles/views/ProfilePage.module.scss';
import { UserContext } from '../context/UserContext';
import { useContext } from 'react';

const ProfilePage = (props: RouteComponentProps) => {
  const { setLoggedIn } = useContext(UserContext);
  const userEncoded = Cookie.get('user', '');
  const removePrefixes =
    userEncoded[0] === 'j' ? userEncoded.substr(2) : userEncoded;
  const userProfileStringify = decodeURIComponent(removePrefixes);
  const userProfile = JSON.parse(userProfileStringify);

  const onLogoutButtonPress = () => {
    setLoggedIn(false);
    Cookie.remove('user');
  };

  return (
    <div className="app">
      <AppHeader title="Profile" />
      <div className="app-main">
        <p className={Styles.secondText}>Username</p>
        <p className={Styles.mainText}>{userProfile.username}</p>
        <p className={Styles.secondText}>Country of Origin</p>
        <p className={Styles.mainText}>{userProfile.country}</p>
        <CTAButton
          className={Styles.logout}
          onButtonClick={onLogoutButtonPress}
        >
          Logout
        </CTAButton>
      </div>
    </div>
  );
};

export default ProfilePage;
