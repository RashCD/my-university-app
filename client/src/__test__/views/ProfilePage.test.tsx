import { render, cleanup } from '@testing-library/react';
import Cookie from '../../util/cookie';
import ProfilePage from '../../views/ProfilePage';

afterEach(cleanup);

describe('Test for profile page', () => {
  test('should set the correct cookie content and show on document', () => {
    const object = { username: 'test', country: 'malaysia' };

    const objectStringify = JSON.stringify(object);

    const objectEncode = encodeURI(objectStringify);

    Cookie.set('user', objectEncode);

    const container = render(<ProfilePage />);

    expect(container.getByText('test')).toBeInTheDocument();

    expect(container.getByText('malaysia')).toBeInTheDocument();
  });

  test('should match snapshot of profile page', () => {
    const container = render(<ProfilePage />);

    expect(container).toMatchSnapshot();
  });
});
