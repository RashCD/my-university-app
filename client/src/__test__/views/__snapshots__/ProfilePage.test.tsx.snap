// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for profile page should match snapshot of profile page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <header
          class="appHeader"
        >
          <h1
            class="title"
          >
            Profile
          </h1>
          <button
            class="button loginButton"
            type="button"
          >
            <span
              class="iconContainer"
              data-icon="true"
              style="--icon-src: url(user.svg); --icon-color: var(--white); --icon-size: 30px;"
            />
            Login
          </button>
        </header>
        <div
          class="app-main"
        >
          <p
            class="secondText"
          >
            Username
          </p>
          <p
            class="mainText"
          >
            test
          </p>
          <p
            class="secondText"
          >
            Country of Origin
          </p>
          <p
            class="mainText"
          >
            malaysia
          </p>
          <button
            class="button cta logout"
            type="button"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <header
        class="appHeader"
      >
        <h1
          class="title"
        >
          Profile
        </h1>
        <button
          class="button loginButton"
          type="button"
        >
          <span
            class="iconContainer"
            data-icon="true"
            style="--icon-src: url(user.svg); --icon-color: var(--white); --icon-size: 30px;"
          />
          Login
        </button>
      </header>
      <div
        class="app-main"
      >
        <p
          class="secondText"
        >
          Username
        </p>
        <p
          class="mainText"
        >
          test
        </p>
        <p
          class="secondText"
        >
          Country of Origin
        </p>
        <p
          class="mainText"
        >
          malaysia
        </p>
        <button
          class="button cta logout"
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
